doc/howtos/OSTree.rst

```diff
@@ -0,0 +1,127 @@
+git, that provides version control and a distribution mechanism for bootable
+OS filesystem trees, or other binaries. For a comprehensive introduction
+to OSTree, an overview of its architecture and feature set, please refer
+to its `documentation <http://ostree.readthedocs.io>_`.
+
+If enabled, OSTree support in refkit
+
+ * provides A/B-versioning (within a single block device partition)
+ * creates a per-image OSTree repository for each image built
+ * exposes builds in a common OSTree repository as a series of commits
+ * can sign each commit (which is then verified during updates)
+ * can provide a service for automatic image updates
+
+
+Enabling OSTree Support
+#######################
+
+To enable end-to-end OSTree support with automatic image updates, you will
+need to
+
+ * enable OSTree support for your builds
+ * ideally generate and use a pair of signing keys for your builds/updates
+ * expose the OSTree repository of your builds over HTTP/HTTPS
+ * point your clients to the exposed OSTree repository
+
+To enable OSTree support, turn on the 'ostree' image feature. You can
+do this by incuding the following configuration snippet in your local.conf
+or other suitable global Yocto configuration file::
+
+    REFKIT_IMAGE_EXTRA_FEATURES += "ostree"
+
+To use your GPG signing key pair for signing and verifying the OSTree
+repository commits, assuming your keys are in the keyrings in <gpg-home>
+with key ID release@example.org, add the following to your local.conf or
+other suitable global Yocto configuration file::
+
+    OSTREE_GPGDIR = "<gpg-home>"
+    OSTREE_GPGID  = "release@example.org"
+
+Assuming you want to use your build machine, build.example.org, in this
+example also as your update server, you can readily point the image for
+updates to your update/build server by adding the following to your
+local.conf or other suitable global Yocto configuration file::
+
+    OSTREE_REMOTE = "http://build.example.org/ostree/"
+
+You can also use HTTPS instead of HTTP if you want to and your server is
+properly configured for serving HTTPS requests.
+
+Next you need to expose the OSTree repository your builds are exported to
+over HTTP/HTTPS for clients to consume. By default this repository is
+located in build/tmp-glibc/deploy/ostree-repo, but you can change this
+location by adding the following to your local.conf or other suitable
+global Yocto configuration file::
+
+    OSTREE_EXPORT = "<path-to-the-repository>"
+
+Now assuming, you did not change the location, and you use Apache for
+service HTTP/HTTPS requests, you can expose this repository with Apache
+by adding the following to your Apache configuration::
+
+    Alias "/ostree/" "<path-to-intel-iot-refkit>/build/tmp-glibc/deploy/ostree-repo/"
+    
+    <Directory <path-to-inte-iot-refkit>/build/tmp-glibc/deploy/ostree-repo>
+        Options Indexes FollowSymLinks
+        Require all granted
+    </Directory>
```

Comment:
Definitely the ostree trivial-http needs to be mentioned and while at it I can mention the python one, too. I'll update the docs.